app/controllers/posts_controller.rb

```diff
@@ -24,6 +24,20 @@ def create
     end
   end
 
+  def destroy
+    @post = Post.find(params[:post_id])
+    authorize! :destroy, @post
+    @comment = @post.comments
+    @comment.each(&:destroy)
+    @post.destroy
+    flash[:success] = ['Post Deleted Successfully']
+
+    respond_to do |format|
+      format.html { redirect_to "/users/#{current_user.id}/posts" }
+      format.json { head :no_content }
+    end
+  end
+
```

Comment:
- [ ] **Required:** Using the `gem cancancan` as mentioned above, this file should be protected against unauthorized posts. Kindly consult the [cancancan documentation.](https://github.com/CanCanCommunity/cancancan#controller-helpers)